File: components/physics/src/main/java/org/storm/physics/collision/Impulsive.java
package org.storm.physics.collision;

import org.storm.physics.entity.Entity;

/**
 * An Impulsive Object is one which reacts when being collided with
 */
public interface Impulsive {

  /**
   * Executes a reaction based on the Entity.
   *
   * @param entity the Entity which collided with this one
   */
  void react(Entity entity);

}

File: components/physics/src/main/kotlin/org/storm/physics/collision/Impulsive.kt
package org.storm.physics.collision

/**
 * An Impulsive Object is one which reacts when being collided with
 */
interface Impulsive {

    /**
     * Executes a reaction based on the Entity.
     *
     * @param collisionObject the Entity which collided with this one
     */
    fun react(collisionObject: CollisionObject)

}
